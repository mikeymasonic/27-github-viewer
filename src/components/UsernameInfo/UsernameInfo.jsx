import React from 'react';
import PropTypes from 'prop-types';

const UsernameInfo = ({ avatar_url, login, followers, following, html_url }) => (
  <figure>
    <img src={avatar_url} alt={login} />
    <figcaption>
      <h4>username:</h4> {login}
      <h4>followers:</h4> {followers}
      <h4>following:</h4> {following}
      <p><a href={html_url} target="_blank" rel="noopener noreferrer">github profile</a></p>
    </figcaption>
  </figure>
);

UsernameInfo.propTypes = {
  avatar_url: PropTypes.string.isRequired,
  login: PropTypes.string.isRequired,
  followers: PropTypes.number.isRequired,
  following: PropTypes.number.isRequired,
  html_url: PropTypes.string.isRequired,
};

export default UsernameInfo;
