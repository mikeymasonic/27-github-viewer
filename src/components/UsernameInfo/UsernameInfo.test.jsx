import React from 'react';
import { shallow } from 'enzyme';
import UsernameInfo from './UsernameInfo';

describe('UsernameInfo component', () => {
  it('renders UsernameInfo', () => {
    const wrapper = shallow(<UsernameInfo 
      avatar_url="https://avatars2.githubusercontent.com/u/35716186?v=4"
      login="mikeymasonic"
      followers={5}
      following={4}
      html_url="https://github.com/mikeymasonic"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
