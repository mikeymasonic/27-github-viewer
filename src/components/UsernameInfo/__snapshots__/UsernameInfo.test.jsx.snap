// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsernameInfo component renders UsernameInfo 1`] = `
<figure>
  <img
    alt="mikeymasonic"
    src="https://avatars2.githubusercontent.com/u/35716186?v=4"
  />
  <figcaption>
    <p>
      mikeymasonic
    </p>
    <p>
      5
    </p>
    <p>
      4
    </p>
    <p>
      <a
        href="https://github.com/mikeymasonic"
        rel="noopener noreferrer"
        target="_blank"
      >
        link
      </a>
    </p>
  </figcaption>
</figure>
`;
